BurnOutSharp/ProtectionType/Macrovision.cs

```diff
@@ -266,6 +268,224 @@ private string CheckSectionForProtection(string file, bool includeDebug, List<st
             return MatchUtil.GetFirstMatch(file, sectionRaw, matchers, includeDebug);
         }
 
+        private List<string> FindProductFromVersion(List<string> resultsList)
```

Comment:
I do see what you're trying to do here but I think it's needlessly complex. I can go over with you what I'm talking about on Discord.